SPDX-FileCopyrightText: 2019 Bruno Haible <bruno@clisp.org>

SPDX-License-Identifier: GPL-3.0-or-later

This is the set of differences between the generated file and hand
modifications made for the file in gettext.

Applied after the regenerations occur.

--- a/include/libxml/globals.h	2025-12-31 10:27:50.432552177 +1100
+++ b/include/libxml/globals.h	2019-12-23 22:21:12.000000000 +1100
@@ -62,6 +91,8 @@
 XMLPUBFUN xmlOutputBufferCreateFilenameFunc
 XMLCALL xmlOutputBufferCreateFilenameDefault (xmlOutputBufferCreateFilenameFunc func);
 
+#if !defined IN_LIBTEXTSTYLE
+
 /*
  * Externally global symbols which need to be protected for backwards
  * compatibility support.
@@ -101,6 +132,8 @@
 #undef  xmlParserInputBufferCreateFilenameValue
 #undef  xmlOutputBufferCreateFilenameValue
 
+#endif
+
 /**
  * xmlRegisterNodeFunc:
  * @node: the current node
--- a/globals.c	2025-12-31 10:42:52.577486689 +1100
+++ b/globals.c	2019-12-23 22:21:12.000000000 +1100
@@ -70,6 +99,8 @@
  *									*
  ************************************************************************/
 
+#if !defined IN_LIBTEXTSTYLE
+
 /*
  * Memory allocation routines
  */
@@ -79,6 +110,8 @@
 #undef	xmlMemStrdup
 #undef	xmlRealloc
 
+#endif
+
 #if defined(DEBUG_MEMORY_LOCATION) || defined(DEBUG_MEMORY)
 xmlFreeFunc xmlFree = (xmlFreeFunc) xmlMemFree;
 xmlMallocFunc xmlMalloc = (xmlMallocFunc) xmlMemMalloc;
@@ -150,6 +183,8 @@
 #include <libxml/globals.h>
 #include <libxml/SAX.h>
 
+#if !defined IN_LIBTEXTSTYLE
+
 #undef	docbDefaultSAXHandler
 #undef	htmlDefaultSAXHandler
 #undef	oldXMLWDcompatibility
@@ -179,6 +214,9 @@
 
 #undef  xmlParserInputBufferCreateFilenameValue
 #undef  xmlOutputBufferCreateFilenameValue
+
+#endif
+
 /**
  * xmlParserVersion:
  *
@@ -702,7 +740,9 @@
 }
 
 #ifdef LIBXML_DOCB_ENABLED
+#if !defined IN_LIBTEXTSTYLE
 #undef	docbDefaultSAXHandler
+#endif
 xmlSAXHandlerV1 *
 __docbDefaultSAXHandler(void) {
     if (IS_MAIN_THREAD)
@@ -713,7 +753,9 @@
 #endif
 
 #ifdef LIBXML_HTML_ENABLED
+#if !defined IN_LIBTEXTSTYLE
 #undef	htmlDefaultSAXHandler
+#endif
 xmlSAXHandlerV1 *
 __htmlDefaultSAXHandler(void) {
     if (IS_MAIN_THREAD)
@@ -723,7 +765,9 @@
 }
 #endif
 
+#if !defined IN_LIBTEXTSTYLE
 #undef xmlLastError
+#endif
 xmlError *
 __xmlLastError(void) {
     if (IS_MAIN_THREAD)
@@ -738,7 +782,9 @@
  * the right place for them :-)
  */
 #if defined(LIBXML_THREAD_ALLOC_ENABLED) && defined(LIBXML_THREAD_ENABLED)
+#if !defined IN_LIBTEXTSTYLE
 #undef xmlMalloc
+#endif
 xmlMallocFunc *
 __xmlMalloc(void){
     if (IS_MAIN_THREAD)
@@ -747,7 +793,9 @@
 	return (&xmlGetGlobalState()->xmlMalloc);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef xmlMallocAtomic
+#endif
 xmlMallocFunc *
 __xmlMallocAtomic(void){
     if (IS_MAIN_THREAD)
@@ -756,7 +804,9 @@
         return (&xmlGetGlobalState()->xmlMallocAtomic);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef xmlRealloc
+#endif
 xmlReallocFunc *
 __xmlRealloc(void){
     if (IS_MAIN_THREAD)
@@ -765,7 +815,9 @@
         return (&xmlGetGlobalState()->xmlRealloc);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef xmlFree
+#endif
 xmlFreeFunc *
 __xmlFree(void){
     if (IS_MAIN_THREAD)
@@ -791,7 +843,9 @@
  */
 
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	oldXMLWDcompatibility
+#endif
 int *
 __oldXMLWDcompatibility(void) {
     if (IS_MAIN_THREAD)
@@ -800,7 +854,9 @@
 	return (&xmlGetGlobalState()->oldXMLWDcompatibility);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlBufferAllocScheme
+#endif
 xmlBufferAllocationScheme *
 __xmlBufferAllocScheme(void) {
     if (IS_MAIN_THREAD)
@@ -817,7 +873,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlDefaultBufferSize
+#endif
 int *
 __xmlDefaultBufferSize(void) {
     if (IS_MAIN_THREAD)
@@ -834,7 +892,10 @@
     return ret;
 }
 
+#ifdef LIBXML_SAX1_ENABLED
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlDefaultSAXHandler
+#endif
 xmlSAXHandlerV1 *
 __xmlDefaultSAXHandler(void) {
     if (IS_MAIN_THREAD)
@@ -842,8 +903,11 @@
     else
 	return (&xmlGetGlobalState()->xmlDefaultSAXHandler);
 }
+#endif /* LIBXML_SAX1_ENABLED */
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlDefaultSAXLocator
+#endif
 xmlSAXLocator *
 __xmlDefaultSAXLocator(void) {
     if (IS_MAIN_THREAD)
@@ -852,7 +916,9 @@
 	return (&xmlGetGlobalState()->xmlDefaultSAXLocator);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlDoValidityCheckingDefaultValue
+#endif
 int *
 __xmlDoValidityCheckingDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -869,7 +935,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlGenericError
+#endif
 xmlGenericErrorFunc *
 __xmlGenericError(void) {
     if (IS_MAIN_THREAD)
@@ -878,7 +946,9 @@
 	return (&xmlGetGlobalState()->xmlGenericError);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlStructuredError
+#endif
 xmlStructuredErrorFunc *
 __xmlStructuredError(void) {
     if (IS_MAIN_THREAD)
@@ -887,7 +957,9 @@
 	return (&xmlGetGlobalState()->xmlStructuredError);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlGenericErrorContext
+#endif
 void * *
 __xmlGenericErrorContext(void) {
     if (IS_MAIN_THREAD)
@@ -896,7 +968,9 @@
 	return (&xmlGetGlobalState()->xmlGenericErrorContext);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlStructuredErrorContext
+#endif
 void * *
 __xmlStructuredErrorContext(void) {
     if (IS_MAIN_THREAD)
@@ -905,7 +979,9 @@
 	return (&xmlGetGlobalState()->xmlStructuredErrorContext);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlGetWarningsDefaultValue
+#endif
 int *
 __xmlGetWarningsDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -922,7 +998,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlIndentTreeOutput
+#endif
 int *
 __xmlIndentTreeOutput(void) {
     if (IS_MAIN_THREAD)
@@ -939,7 +1017,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlTreeIndentString
+#endif
 const char * *
 __xmlTreeIndentString(void) {
     if (IS_MAIN_THREAD)
@@ -956,7 +1036,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlKeepBlanksDefaultValue
+#endif
 int *
 __xmlKeepBlanksDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -973,7 +1055,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlLineNumbersDefaultValue
+#endif
 int *
 __xmlLineNumbersDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -990,7 +1074,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlLoadExtDtdDefaultValue
+#endif
 int *
 __xmlLoadExtDtdDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -1007,7 +1093,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlParserDebugEntities
+#endif
 int *
 __xmlParserDebugEntities(void) {
     if (IS_MAIN_THREAD)
@@ -1024,7 +1112,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlParserVersion
+#endif
 const char * *
 __xmlParserVersion(void) {
     if (IS_MAIN_THREAD)
@@ -1033,7 +1123,9 @@
 	return (&xmlGetGlobalState()->xmlParserVersion);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlPedanticParserDefaultValue
+#endif
 int *
 __xmlPedanticParserDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -1050,7 +1142,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlSaveNoEmptyTags
+#endif
 int *
 __xmlSaveNoEmptyTags(void) {
     if (IS_MAIN_THREAD)
@@ -1067,7 +1161,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlSubstituteEntitiesDefaultValue
+#endif
 int *
 __xmlSubstituteEntitiesDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -1084,7 +1180,9 @@
     return ret;
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlRegisterNodeDefaultValue
+#endif
 xmlRegisterNodeFunc *
 __xmlRegisterNodeDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -1093,7 +1191,9 @@
 	return (&xmlGetGlobalState()->xmlRegisterNodeDefaultValue);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlDeregisterNodeDefaultValue
+#endif
 xmlDeregisterNodeFunc *
 __xmlDeregisterNodeDefaultValue(void) {
     if (IS_MAIN_THREAD)
@@ -1102,7 +1202,9 @@
 	return (&xmlGetGlobalState()->xmlDeregisterNodeDefaultValue);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlParserInputBufferCreateFilenameValue
+#endif
 xmlParserInputBufferCreateFilenameFunc *
 __xmlParserInputBufferCreateFilenameValue(void) {
     if (IS_MAIN_THREAD)
@@ -1111,7 +1213,9 @@
 	return (&xmlGetGlobalState()->xmlParserInputBufferCreateFilenameValue);
 }
 
+#if !defined IN_LIBTEXTSTYLE
 #undef	xmlOutputBufferCreateFilenameValue
+#endif
 xmlOutputBufferCreateFilenameFunc *
 __xmlOutputBufferCreateFilenameValue(void) {
     if (IS_MAIN_THREAD)
@@ -1119,3 +1223,6 @@
     else
 	return (&xmlGetGlobalState()->xmlOutputBufferCreateFilenameValue);
 }
+
+#define bottom_globals
+#include "elfgcchack.h"
